### PR TITLE
Daml-LF: Damlification of Scala/Haskell files

### PR DESCRIPTION
**Pull request by miklos-da**
Made references to Daml-LF consistent with our product naming conventions.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM.

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala [hunk 1]
```diff
@@ -262,18 +262,18 @@ private[data] final class IdStringImpl extends IdString {
     @throws[IllegalArgumentException]
     override def assertFromString(s: String) = {
       if (s.length == 0)
-        throw new IllegalArgumentException("DAML LF Name is empty")
+        throw new IllegalArgumentException("Daml LF Name is empty")
```
Comment (remyhaemmerle-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala:265]:
```suggestion
        throw new IllegalArgumentException("Daml-LF Name is empty")
```
We probably should use the occasion, to make "Daml LF" and Daml-LF" consistent.

Comment (miklos-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala]: Good point -- ran a search & replace to do that.

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala [hunk 2]
```diff
@@ -81,7 +81,7 @@ sealed abstract class IdString {
   type Party <: String
 
   /** Reference to a package via a package identifier. The identifier is the ascii7
-    * lowercase hex-encoded hash of the package contents found in the Daml LF Archive.
+    * lowercase hex-encoded hash of the package contents found in the Daml-LF Archive.
```
Comment (remyhaemmerle-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala:84]: :bow:

**Review by stefanobaghino-da — APPROVED** [review]
~LGTM~ Lgtm.

**Pull request merged**